solr/core/src/test/org/apache/solr/cloud/ReplaceNodeTest.java

```diff
@@ -115,9 +115,9 @@ public void test() throws Exception {
     log.info("excluded_node : {}  ", emptyNode);
     createReplaceNodeRequest(nodeToBeDecommissioned, emptyNode, null)
         .processAndWait("000", cloudClient, 15);
+    ZkStateReader zkStateReader = ZkStateReader.from(cloudClient);
```

Comment:
Does this zkStateReader need to be closed?

Reply:
No; it was simply extracted from the CloudSolrClient's `ZkClientClusterStateProvider`

Reply:
The method name does make it sound like it's possibly creating a new instance. Maybe renaming the method to `extractFrom` could be helpful? Not a deal breaker, just could be a bit easier to understand.

Reply:
Note this particular choice was established in SOLR-16061 in March.  Renaming this method would be a separate issue.  `extractFrom` seems fine; perhaps even shorter, just `get`?

Reply:
Does it hurt to close the ZkStateReader? Wrap in try-with-resources?

```
try (ZkStateReader zkStateReader = ZkStateReader.from(cloudClient)) {
...
}
```

I'm not that worried about the name, but want to make sure if things are closable they are closed.

Reply:
We shouldn't do that.  `ZkStateReader.from` is a glorified "get" but keeps ZooKeeper dependencies off of CloudSolrClient.

Reply:
Not that SolrClient is already Closable and we should already be taking care to manage their lifecycle.  A CloudSolrClient that uses a ZK state provider will close it.